src/common/internal/query/json_param_value.ts

```diff
@@ -42,12 +45,21 @@ function stringifyFilter(k: string, v: unknown): unknown {
       v !== jsNegativeZeroMagicValue,
       `${v} is a magic value for stringification, so cannot be used`
     );
+
+    assert(
+      v.match(jsBigIntMagicPattern) === null,
+      `${v} matches bigint magic pattern for stringification, so cannot be used`
+    );
   }
 
   if (Object.is(v, -0)) {
     return jsNegativeZeroMagicValue;
   }
 
+  if (typeof v === 'bigint') {
+    return `_${v}n_`;
```

Comment:
Do we need the underscores? Is `(\d+)n` not sufficient?

Reply:
I'm fine with no underscores, since we'll detect if a collision ever happens.

Reply:
Also good catch on the `^$`.